src/services/challenges_repository.js

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright: Ambrosus Technologies GmbH
+Email: tech@ambrosus.com
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+import {getDefaultAddress} from '../utils/web3_tools';
+
+export default class ChallengesRepository {
+  constructor(contractManager) {
+    this.contractManager = contractManager;
+  }
+
+  async previousChallenges(fromBlock = 0) {
+    const challengesContract = await this.contractManager.challengesContract();
+    const events = await challengesContract.getPastEvents('ChallengeCreated', {fromBlock});
+    return {
+      challenges: events.map(({returnValues: {sheltererId, bundleId, challengeId}}) => ({sheltererId, bundleId, challengeId})),
+      maxBlockNumber: events.reduce((maxBlock, {blockNumber}) => Math.max(maxBlock, blockNumber), 0)
```

Comment:
We may want to use getBlock instead but it leaves a chance that getting block and calling getPastEvents happen on different blocks, as a result we either miss some challenges or get them twice